nessie/lakectl_test.go

```diff
@@ -295,3 +301,52 @@ func TestLakectlAuthUsers(t *testing.T) {
 	// Cleanup
 	RunCmdAndVerifySuccess(t, Lakectl()+" auth users delete --id "+userName, false, "User deleted successfully\n", vars)
 }
+
+func TestLakectlIngest(t *testing.T) {
+	repoName := generateUniqueRepositoryName()
+	storage := generateUniqueStorageNamespace(repoName)
+	tempBucketName := "ingest-data-" + repoName
+	vars := map[string]string{
+		"REPO":    repoName,
+		"STORAGE": storage,
+		"BRANCH":  mainBranch,
+	}
+
+	// set-up: creating a dummy bucket and uploading some files into it. Later this bucket will be used as the "from"
+	// location. This is done as the `lakectl ingest` commad runs with local aws settings and can only access the
+	// corresponding region
+
+	s3Client, err := store.GetS3Client()
+	require.NoError(t, err, "Failed to initialize S3 session")
+
+	_, err = s3Client.CreateBucket(&s3.CreateBucketInput{Bucket: aws.String(tempBucketName)})
+	require.NoError(t, err, "Failed to create bucket", tempBucketName)
+
+	defer cleanS3Bucket(s3Client, tempBucketName)
```

Comment:
Creating a bucket per run can blow up if deleting the bucket doesn't work as expected, or the test is stopped half way through.

I suggest instead to use a predefined bucket and use prefixes for the ingestion and the repo part.
For example `s3://test-bucket-for-lakectl-ingestion/` where the prefix ingestion `ingestion-raw-run-xxxxx` and storage namespace for the matching repo `ingestion-repo-run-xxxxx`. Then this bucket can have lifecycle policies for cleanups (let's say 2 weeks) and you don't have to worry about dangling buckets.

You can also use the bucket `s3://nessie-system-testing-data` for the raw to-be-ingested data. That bucket data isn't deleted and you won't have to upload these files yourself.